django/db/models/lookups.py

```diff
@@ -538,12 +538,18 @@ class IRegex(Regex):
 
 
 class YearLookup(Lookup):
+    iso_year = False
+
     def year_lookup_bounds(self, connection, year):
         output_field = self.lhs.lhs.output_field
         if isinstance(output_field, DateTimeField):
-            bounds = connection.ops.year_lookup_bounds_for_datetime_field(year)
+            bounds = connection.ops.year_lookup_bounds_for_datetime_field(
+                year, iso_year=self.iso_year,
+            )
         else:
-            bounds = connection.ops.year_lookup_bounds_for_date_field(year)
+            bounds = connection.ops.year_lookup_bounds_for_date_field(
+                year, iso_year=self.iso_year,
+            )
```

Comment:
I would remove extra lookups and based `iso_year` on the `self.lhs`, e.g.:
```python
    def year_lookup_bounds(self, connection, year):
        from django.db.models.functions import ExtractIsoYear
        
        iso_year = isinstance(self.lhs, ExtractIsoYear)
        output_field = self.lhs.lhs.output_field
        if isinstance(output_field, DateTimeField):
            bounds = connection.ops.year_lookup_bounds_for_datetime_field(
                year, iso_year=iso_year,
            )
        else:
            bounds = connection.ops.year_lookup_bounds_for_date_field(
                year, iso_year=iso_year,
            )
        return bounds
```

Reply:
clever, done: 303949dbb343465671256f5c2f0e49db92affb48